conda_auth/hooks.py

```diff
@@ -1,36 +1,47 @@
 """
 A place to register plugin hooks
 """
-from conda.plugins import CondaAuthHandler, CondaPreCommand, hookimpl
+from conda.plugins import CondaAuthHandler, CondaPreCommand, CondaSubcommand, hookimpl
 
-from . import basic_auth, oauth2
+from .handlers import OAuth2Handler, BasicAuthHandler, oauth2, basic_auth
 from .constants import OAUTH2_NAME, HTTP_BASIC_AUTH_NAME
+from .cli import login_wrapper, logout_wrapper
+
+
+@hookimpl
+def conda_subcommands():
+    """
+    Registers subcommands
+    """
+    yield CondaSubcommand(
+        name="login", action=login_wrapper, summary="Login to a channel"
+    )
+    yield CondaSubcommand(
+        name="logout", action=logout_wrapper, summary="Logout of a channel"
+    )
```

Comment:
Have you thought about making these commands of a common `auth` subcommand (`conda auth login`)? That might make it easier to scale the user experience and group additional subcommands in the future?

Reply:
From a namespace perspective, that makes sense, but from a user experience perspective, it's annoying to have to type out `conda auth login <channel>` when you need to login in.

I'm a little torn about what to do... We could always just introduce a `conda auth` namespace for other, less used commands in the future.

Reply:
This is a good compromise, cheers!

Reply:
I think we need to consider other reasons users may need to login/logout of conda before allocating such valuable subcommands.

For example, what if the `anaconda-client` is refactored into conda subcommands? Would we expect them to also use the login/logout subcommands? Would we expect all of their commands to be nested (e.g., `conda anaconda login`)?

I'm also thinking that `conda login` suggests to the user that they are logging into conda itself, which is not what is happening here. What we are actually doing here is `conda channel login`.

Reply:
@kenodegard,

I think what you're hinting at here is the the "top level" command namespace is quite valuable for conda, and we need a set of best practices for meaningfully allocating it. The thing that makes the most sense for me is that a plugin's namespace would mirror what the package name is (for `conda-auth` it would be `auth`).

Ergonomically, this kind of sucks for end users who have to type this out, but I recently came across something in another complex CLI tool that I think could help out: https://cli.github.com/manual/gh_alias

In the GitHub CLI, there is an "alias" command that you can use to make shortcuts. This could be a useful measure for us to put in place because as we continue to add new subcommands from different plugins, this will increasingly become a problem (e.g. `conda somethinglong subcommand sub-subcommand --long-option <holy-crap-heres-the-argument>`)

If we could agree on implementing such a feature for conda, I would be more than happy to go with `conda auth login/logout`.

Reply:
> I think what you're hinting at here is the the "top level" command namespace is quite valuable for conda, and we need a set of best practices for meaningfully allocating it.

Exactly and 💯

> Ergonomically, this kind of sucks for end users who have to type this out

Yes, but we also need to think about how ofter a user will be typing this. Will a user be logging in/out hourly? Daily? Weekly? Monthly?

> In the GitHub CLI, there is an "alias" command that you can use to make shortcuts.

Yeah same in `git`, totally down for making this happen.